### PR TITLE
macos-pkg: apply required entitlements to vfkit binary during pkg build

### PR DESCRIPTION
**Pull request by anjannath**
although the required entitlements are already applied to the released vfkit
binary, they gets overwritten when codesiging again during the pkg build
process, so we download the entitlements file from code-ready/vfkit and re-apply

### CODE REVIEW

File: packaging/package.sh [hunk 1]
```diff
@@ -3,11 +3,15 @@ set -euxo pipefail
 
 BASEDIR=$(dirname "$0")
 OUTPUT=$1
-GOARCH=$(go env GOARCH)
+GOARCH=amd64
```
Comment (anjannath) [inline packaging/package.sh:6]: i kept the name as `GOARCH`. but if others think its better to change then i'll update to `MARCH` or just `ARCH`

**Comment by anjannath** [comment]
Note: waiting for a signed pkg to test from rcm

**Comment by anjannath** [comment]
> The biggest question is why are we always resigning everything? Ideally, we'd only sign vfkit once per vfkit release, and reuse the signed binary in crc, same for admin-helper.

indeed, It'd be better if while doing the `vfkit` release we can publish signed binaries, (its just that we have to follow the whole rcm process) provided the releases of `vfkit`, `admin-helper` will be of lesser frequency this is something we should look into

File: packaging/package.sh (hunk 1 above)
Comment (anjannath) [reply packaging/package.sh]: Updated! PTALA

File: packaging/package.sh [hunk 2]
```diff
@@ -3,11 +3,23 @@ set -euxo pipefail
...
 OUTPUT=$1
-GOARCH=$(go env GOARCH)
 CODESIGN_IDENTITY=${CODESIGN_IDENTITY:-mock}
 PRODUCTSIGN_IDENTITY=${PRODUCTSIGN_IDENTITY:-mock}
 NO_CODESIGN=${NO_CODESIGN:-0}
 
+case "$(uname -m)" in
+    "x86_64")
+         GOARCH="amd64"
+        ;;
```
Comment (cfergeau) [inline packaging/package.sh:13]: Indentation might be a little bit off here?

**Review by cfergeau — APPROVED** [review]

File: packaging/package.sh (hunk 2 above)
Comment (anjannath) [reply packaging/package.sh:13]: yeah, will fix this and merge

**Pull request merged**